wandb/apis/public/runs.py

```diff
@@ -952,15 +1040,55 @@ def log_artifact(
         )
         return artifact
 
+    def load_full_data(self, force=False):
+        """Load full run data including heavy fields like config, systemMetrics, summaryMetrics.
+
+        This method is useful when you initially used lightweight=True for listing runs,
+        but need access to the full data for specific runs.
+
+        Args:
+            force (bool): Force reload even if data is already loaded
+        """
+        if not self._lightweight and not force:
+            # Already in full mode, no need to reload
+            return self._attrs
+
+        # Temporarily switch to full mode for this load
+        original_lightweight = self._lightweight
+        self._lightweight = False
```

Comment:
Why not make `lightweight` an argument to `self.load()`? If we can't modify the signature of `self.load()`, we can always extract its implementation to a new private method.